### PR TITLE
fixed issue #53

### PR DESCRIPTION
**Pull request by whitelok**
fixed issue #53

### CODE REVIEW

File: tumblr-photo-video-ripper.py [hunk 1]
```diff
@@ -8,6 +8,8 @@
 from threading import Thread
 import re
 import json
+import defusedexpat
```
Comment (dixudx) [inline tumblr-photo-video-ripper.py:11]: This is a new package. Please add to `requirements.txt` as well.

File: tumblr-photo-video-ripper.py [hunk 2]
```diff
@@ -214,6 +218,9 @@ def _download_media(self, site, medium_type, start):
             except UnicodeDecodeError:
                 print("Cannot decode response data from URL %s" % media_url)
                 continue
+            except ExpatError:
```
Comment (dixudx) [inline tumblr-photo-video-ripper.py:221]: To use this `ExpatError`, you have to import `from xml.parsers.expat import ExpatError` first.

File: tumblr-photo-video-ripper.py [hunk 3]
```diff
@@ -214,6 +218,9 @@ def _download_media(self, site, medium_type, start):
...
+                print("Cannot decode response data from URL %s" % media_url)
```
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: This message should be slightly different from above `UnicodeDecodeError`.

**Review by dixudx — CHANGES_REQUESTED** [review]
Thanks for your contribution.
A few comments. Others lgtm.

**Comment by whitelok** [comment]
Thanks for your notification.
All enhancement is in the latest commit (b9c2023707ddf5c3ed17603c85b0654acd765c84)

File: tumblr-photo-video-ripper.py (hunk 1 above)
Comment (dixudx) [reply tumblr-photo-video-ripper.py:11]: This is still not addressed yet. Please update it.

**Comment by dixudx** [comment]
@whitelok Please squash your commits as well.

**Comment by whitelok** [comment]
Done on 081573ec77af305b35be97827c74d6c29a9fc4c9
Thanks for your review.

File: tumblr-photo-video-ripper.py [hunk 4]
```diff
@@ -196,7 +198,9 @@ def _download_media(self, site, medium_type, start):
                 break
 
             try:
-                data = xmltodict.parse(response.content)
+                xml_cleaned = re.sub(u'[^\n\r\t\x20-\x7f]+',
```
Comment (dixudx) [inline tumblr-photo-video-ripper.py]: I was wondering whether we should ignore `\n`, `\r` and `\t`, since they are not the non-ASCII characters.

Comment (whitelok) [reply tumblr-photo-video-ripper.py]: While I debug the xml from tumblr, some attributes have '\n', '\r', '\t'. Hence, I add this 3 for avoid xml parser error. But your suggestion is right, I would like to test whether is it ok without '\n', '\r', '\t'. Thanks

File: tumblr-photo-video-ripper.py [hunk 5]
```diff
@@ -196,7 +198,9 @@ def _download_media(self, site, medium_type, start):
                 break
 
             try:
-                data = xmltodict.parse(response.content)
+                xml_cleaned = re.sub(u'[^\x20-\x7f]+',
```
Comment (whitelok) [inline tumblr-photo-video-ripper.py:201]: Testing pass. You are right. \n \t \r will not effect the downloading result.
Thanks

**Comment by dixudx** [comment]
@whitelok Thanks for your contribution. Good work!

**Pull request merged**